### PR TITLE
react-docgen@5.0.0 upgrade which adds typescript support out of box

### LINKED ISSUE #1354: Upgrade react-docgen to 5.0.0 beta
[A new version](https://github.com/reactjs/react-docgen/releases/tag/v5.0.0-beta.1) has TypeScript support, so no need to extra plugins anymore. 

- [ ] Use `tsType` similar to `flowType`.
- [ ] Update docs, remove all mentions to react-docgen-typescript.

**This is a breaking change.** Should be based on #1288.

### PR DESCRIPTION
**Pull request by thecodejack**
continuation of https://github.com/styleguidist/react-styleguidist/pull/1387
Handled @sapegin comments of https://github.com/styleguidist/react-styleguidist/pull/1387

Closes #1354

### CODE REVIEW

File: docs/Thirdparties.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ Styleguidist _loads_ your components (see [Loading and exposing components](Comp
 
 React-docgen reads your components as static text files and looks for patterns like class or function declarations that look like React components. It does not run any JavaScript code, so, if your component is dynamically generated, is wrapped in a higher-order component, or is split into several files, then react-docgen may not understand it.
 
-It supports components defined via `React.createClass`, ES6 classes and stateless functional components, with optional Flow and TypeScript (via [react-docgen-typescript](https://github.com/styleguidist/react-docgen-typescript)) type annotations.
+It supports components defined via `React.createClass`, ES6 classes and stateless functional components, with optional Flow and TypeScript type annotations.
```
Comment (sapegin) [inline docs/Thirdparties.md]: Let's use the official name while we're here:

```suggestion
It supports components defined via `React.createClass`, ES6 classes and function components, with optional Flow and TypeScript type annotations.
```

File: examples/styled-components/Readme.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Example of React Styleguidist using styled-components(and emotion) with typescipt
```
Comment (sapegin) [inline examples/styled-components/Readme.md]:
```suggestion
# Example of React Styleguidist using styled-components (and Emotion) with TypeScript
```

File: examples/styled-components/src/components/Heading/Heading.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+```jsx harmony
```
Comment (sapegin) [inline examples/styled-components/src/components/Heading/Heading.md]:
```suggestion
```jsx
```

File: examples/styled-components/styleguide.config.js [hunk 4]
```diff
@@ -2,17 +2,15 @@ const path = require('path');
 const { version } = require('./package');
 
 module.exports = {
-	components: 'src/components/**/index.ts',
+	components: 'src/components/**/*.{js,tsx}',
```
Comment (sapegin) [inline examples/styled-components/styleguide.config.js]: Can you try to remove this line? The default pattern should work here.

Comment (thecodejack) [reply examples/styled-components/styleguide.config.js]: I did try..components are coming up twice

image

I can fix naming if you still want me to remove the config.

File: docs/Thirdparties.md (hunk 1 above)
Comment (thecodejack) [reply docs/Thirdparties.md]: done

File: examples/styled-components/Readme.md (hunk 2 above)
Comment (thecodejack) [reply examples/styled-components/Readme.md]: done

File: examples/styled-components/src/components/Heading/Heading.md (hunk 3 above)
Comment (thecodejack) [reply examples/styled-components/src/components/Heading/Heading.md]: done

**Comment by thecodejack** [comment]
sorry for responding late. Worked on suggestions and pushed changes.

File: examples/styled-components/styleguide.config.js (hunk 4 above)
Comment (okonet) [reply examples/styled-components/styleguide.config.js]: Is this a bug in Styleguidist?

Comment (thecodejack) [reply examples/styled-components/styleguide.config.js]: Probably..if you have a component folder structure where there are index file and component named file. Styleguidist should have considered only one of them.

**Pull request merged**